app/services/meta-tags.ts

```diff
@@ -0,0 +1,184 @@
+import { service } from '@ember-decorators/service';
+import Service from '@ember/service';
+import config from 'ember-get-config';
+import pathJoin from 'ember-osf-web/utils/path-join';
+
+export type NameAttr = 'name' | 'property';
+
+export type Content = string | number | null | undefined;
+
+export type DataContent = Content | Content[];
+
+/* eslint-disable camelcase */
+export interface MetaTagsData {
+    title: DataContent;
+    type?: DataContent;
+    description?: DataContent;
+    url?: DataContent;
+    doi?: DataContent;
+    identifier?: DataContent;
+    published_date?: DataContent;
+    modified_date?: DataContent;
+    license?: DataContent;
+    language?: DataContent;
+    image?: DataContent;
+    image_type?: DataContent;
+    image_width?: DataContent;
+    image_height?: DataContent;
+    image_alt?: DataContent;
+    site_name?: DataContent;
+    institution?: DataContent;
+    twitter_site?: DataContent;
+    twitter_creator?: DataContent;
+}
+/* eslint-enable camelcase */
+
+export interface MetaTagsDefs {
+    [s: string]: DataContent;
+}
+
+export interface NameMetaTagAttrs {
+    name: string; // eslint-disable-line no-restricted-globals
+    content: Content;
+}
+
+export interface PropMetaTagAttrs {
+    property: string;
+    content: Content;
+}
+
+export type MetaTagAttrs = NameMetaTagAttrs | PropMetaTagAttrs;
+
+export interface HeadTagDef {
+    type: string;
+    attrs: MetaTagAttrs;
+}
+
+export default class MetaTags extends Service {
+    @service i18n;
+
+    /**
+     * Get meta tag definitions.
+     *
+     * @method getMetaTags
+     * @param {MetaTagsData} metaTagsOverrides Data values to override defaults.
+     * @return {MetaTagsDefs} Returns meta tag definitions.
+     */
+    getMetaTags(this: MetaTags, metaTagsOverrides: MetaTagsData): MetaTagsDefs {
+        // Default values.
+        const metaTagsData: MetaTagsData = {
+            title: undefined,
```

Comment:
Might be easier to read quickly if you leave out the ones without a default value.

Reply:
My thinking was that we would add default values for these at some point so I left them in as placeholders. We have the `MetaTagsData` typedef, though, so it's probably not necessary.